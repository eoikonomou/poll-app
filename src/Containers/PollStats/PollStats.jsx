import React from 'react';

const PollStats = () => (
  <div className="poll-stats">
    Poll Stats section
  </div>
);

PollStats.displayName = 'PollStats';

PollStats.propTypes = {};

export default PollStats;
