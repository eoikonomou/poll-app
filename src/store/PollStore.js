import { action, observable } from 'mobx';

export default class PollStore {
  @observable pollQuestion = '';

  @observable pollOptions = [];

  @observable pollLocked = false;

  @action setPollQuestion = (data) => {
    this.pollQuestion = data;
  }

  @action setPollOptions = (data) => {
    this.pollOptions = data;
  }

  @action addPollOption = (data) => {
    this.pollOptions.push(data);
  }

  @action addVote = (optionId) => {
    const option = this.pollOptions.find((opt) => opt.id === optionId);
    option.count += 1;
  }

  @action reset = () => {
    this.pollOptions = [];
    this.pollQuestion = '';
    this.pollLocked = false;
  }

  @action setPollLocked = (data) => {
    this.pollLocked = data;
  }
}
