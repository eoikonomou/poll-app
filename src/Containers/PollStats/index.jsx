import PollStats from './PollStats';

export default PollStats;
