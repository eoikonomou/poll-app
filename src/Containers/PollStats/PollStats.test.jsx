import React from 'react';
import { shallow } from 'enzyme';
import PollStats from './PollStats';
import PollStore from '../../store/PollStore';

describe('PollStats', () => {
  let store = null;

  beforeEach(() => {
    store = new PollStore();
  });

  afterEach(() => {
    store = null;
  });

  test('should render successfully', () => {
    const wrapper = shallow(<PollStats store={store} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render with default title if pollQuestion is empty', () => {
    const wrapper = shallow(<PollStats store={store} />);
    const title = () => wrapper.find('.section-title').at(0);
    expect(title().text()).toEqual('Poll Question');
  });

  test('should change the title when pollQuestion changes', () => {
    store.setPollQuestion('test');
    const wrapper = shallow(<PollStats store={store} />);
    const title = () => wrapper.find('.section-title').at(0);
    expect(title().text()).toEqual('test');
  });
});
