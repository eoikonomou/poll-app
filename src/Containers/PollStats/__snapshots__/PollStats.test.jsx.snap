// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollStats should render successfully 1`] = `
<div
  className="section"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": 2,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <div
    className="section-title"
  >
    Poll Question
  </div>
  <Chart
    data={Array []}
    height={450}
    width={450}
    xDataKey="name"
    yDataKey="value"
  />
</div>
`;
